packages/common/src/utils/intent.ts

```diff
@@ -57,6 +57,7 @@ export interface JSONEntity {
   startPos: number;
   endPos: number;
   entity: string;
+  key: string;
```

Comment:
This might need a more descriptive variable name or at least a comment with this form

```js
/**
 * This key is responsible for so-and-so
 */
```

It's kinda hard to tell what this is and why you need it.